### PR TITLE
[dif] Add tests for dif_alert_handler.h

### PR DESCRIPTION
**Pull request by mcy**
Still pending >16 alert support, I suspect that will be its own PR.

### CODE REVIEW

**Review by gkelly — COMMENTED** [review]
I think the tests look good, just need to split the commit (and it's a separate file, so ezpz) and it's good to go.

File: sw/device/lib/dif/dif_alert_handler.c [hunk 1]
```diff
@@ -933,13 +933,13 @@ dif_alert_handler_result_t dif_alert_handler_get_class_state(
...
     case ALERT_HANDLER_CLASSA_STATE_CLASSA_STATE_VALUE_PHASE2:
-      *state = kDifAlertHandlerClassStatePhase0;
+      *state = kDifAlertHandlerClassStatePhase2;
       break;
     case ALERT_HANDLER_CLASSA_STATE_CLASSA_STATE_VALUE_PHASE3:
-      *state = kDifAlertHandlerClassStatePhase0;
+      *state = kDifAlertHandlerClassStatePhase3;
       break;
     case ALERT_HANDLER_CLASSA_STATE_CLASSA_STATE_VALUE_TERMINAL:
       *state = kDifAlertHandlerClassStateTerminal;
```
[the remark this answers is not in the document]
Comment (gkelly) [reply sw/device/lib/dif/dif_alert_handler.c:945]: This comment covers the whole file and header.

[the remark this answers is not in the document]
Comment (mcy) [reply sw/device/lib/dif/dif_alert_handler.c:945]: Should be split now.

**Review by gkelly — APPROVED** [review]

File: sw/device/tests/dif/dif_alert_handler_unittest.cc [hunk 2]
```diff
@@ -0,0 +1,922 @@
...
+
+  // Disables expectations on MMIO operations. This is useful for configuration
+  // tests that partially configure the device but fail due to a configuration
+  // error, returning kError.
+  void IgnoreMmioCalls() {
+    dev_ = std::make_unique<testing::NiceMock<MockDevice>>();
+    handler_.params.base_addr = dev().region();
+
+    // Make sure that the peripheral looks unlocked.
+    ON_CALL(*dev_, Read32(_)).WillByDefault(Return(kAllOnes));
```
Comment (alphan) [inline sw/device/tests/dif/dif_alert_handler_unittest.cc:92]: This made me wish that `InSequence` in `MmioTest` was opt-out. I gave `Times(AnyNumber())` a try (while keeping the mock strict) but it didn't work.

Kudos for not writing MMIO expectations for calls that return `k...Error`.

**Review by alphan — APPROVED** [review]
Thanks @mcy, LGTM.

File: sw/device/tests/dif/dif_alert_handler_unittest.cc (hunk 2 above)
Comment (mcy) [reply sw/device/tests/dif/dif_alert_handler_unittest.cc:92]: I think it might not be unreasonable to generalize your un-sequenced mixin!

**Comment by mcy** [comment]
Most recent CI run is green: https://dev.azure.com/lowrisc/opentitan/_build/results?buildId=24540&view=results, merging.

**Pull request merged**